lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -85,13 +88,19 @@ def get_collection_versions(self, requirement):
         # type: (Requirement) -> Iterable[Tuple[str, GalaxyAPI]]
         """Get a set of unique versions for FQCN on Galaxy servers."""
         if requirement.is_concrete_artifact:
-            return {
-                (
-                    self._concrete_art_mgr.
-                    get_direct_collection_version(requirement),
-                    requirement.src,
-                ),
-            }
+            version = self._concrete_art_mgr.get_direct_collection_version(requirement)
+
+            err = f"Invalid version found for requirement '{requirement}': {version} ({type(version)}). "
+            err += "The version should be a semantic version string (or '*')."
+            if not isinstance(version, string_types):
+                raise AnsibleError(err)
+            elif version != "*":
+                try:
+                    SemanticVersion(version)
+                except ValueError:
+                    raise AnsibleError(err)
```

Comment:
Mind using `six.raise_from()` here?
```suggestion
                except ValueError as val_err:
                    raise raise_from(AnsibleError(err), val_err)
```

Reply:
Oh, I forgot that we can use native syntax now:
```suggestion
                except ValueError as val_err:
                    raise AnsibleError(err) from val_err
```

Reply:
Nice, good catch.